datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxBackupRestoreService.java

```diff
@@ -695,4 +696,64 @@ private static boolean isVersionEqual(SdxCluster cluster, String baseVersion) {
+            if (backupStatus != null && isRunningState(backupStatus.getState())) {
+                LOGGER.error("Found a backup in progress on {} with status {}", sdxClusterName, backupStatus.getState());
+                return true;
+            }
+            LOGGER.info("No backup in progress found on {}. Good to continue.", sdxClusterName);
+            return false;
+        } catch (RuntimeException exception) {
+            if (exception.getMessage().contains("Status information for backup operation on datalake")
+                    && exception.getMessage().contains("not found")) {
+                // If the RuntimeException's error message matches the message that was thrown when no backup/restore status found, the program continues because
+                // it means this datalake has never run a backup or restore.
+                LOGGER.info("No backup ever happened on {}. Good to continue.", sdxClusterName);
+                return false;
+            } else {
+                LOGGER.error("A RuntimeException happened when trying to get datalake backup status.");
+                throw exception;
+            }
+        }
+    }
+
+    public boolean isDatalakeInRestoreProgress(String sdxClusterName, String actorCrn) {
+        if (sdxClusterName == null || actorCrn == null) {
+            LOGGER.error("Didn't find datalake name or actor CRN. Can not validate if the datalake is in restore. Continue to next step.");
+            return false;
+        }
+        try {
+            DatalakeRestoreStatusResponse restoreStatus = datalakeDrClient.getRestoreStatus(sdxClusterName, actorCrn);
+            if (restoreStatus != null && isRunningState(restoreStatus.getState())) {
+                LOGGER.error("Found a restore in progress on {} with status {}", sdxClusterName, restoreStatus.getState());
+                return true;
+            }
+            LOGGER.info("No restore in progress found on {}. Good to continue.", sdxClusterName);
+            return false;
+        } catch (RuntimeException exception) {
+            if (exception.getMessage().contains("Status information for restore operation on datalake")
+                    && exception.getMessage().contains("not found")) {
+                LOGGER.info("No restore ever happened on {}. Good to continue.", sdxClusterName);
+                return false;
+            } else {
+                LOGGER.error("A RuntimeException happened when trying to get datalake restore status.");
+                throw exception;
+            }
+        }
+    }
+
+    private boolean isRunningState(DatalakeOperationStatus.State state) {
+        return state != null
+                && (state == DatalakeOperationStatus.State.STARTED
+                || state == DatalakeOperationStatus.State.IN_PROGRESS
+                || state == DatalakeOperationStatus.State.VALIDATION_SUCCESSFUL);
```

Comment:
in case of `--validation-only` flag backup will stop after VALIDATION_SUCCESSFUL, so it is not a running state

Reply:
@votez You are right! Thank you!! Removed that!!